### PR TITLE
ucx: add backtrace-detail variant

### PR DESCRIPTION
**Pull request by tvandera**
ucx has the configure option --[enable|disable]-backtrace detail.
This option is not explicitely set by spack, causing problems on my system, because
./configure does not find the bfd.h header file / libbfd.so library.

Added variant + dependencies (binutils). Disabled by default

### CODE REVIEW

**Comment by alalazo** [comment]
Waiting for @hppritcha to review before merging.

**Comment by hppritcha** [comment]
Looks good to me!

**Review by alalazo — APPROVED** [review]

**Pull request merged**
